lib/rule-helpers.js

```diff
@@ -21,11 +20,15 @@ export function takeWhile(source, predicate) {
 
 // Retrieves style guide documentation with cached responses
 export async function getRuleMarkDown(rule) {
-  if (new Date().getTime() - docsLastRetrieved < DOCUMENTATION_LIFETIME) {
-    // If documentation is stale, clear cache
-    docsRuleCache.clear();
+  if (docsRuleCache.has(rule)) {
+    const cachedRule = docsRuleCache.get(rule);
+    if (new Date().getTime() - cachedRule.lasFetched < DOCUMENTATION_LIFETIME) {
```

Comment:
Should not be better use `expires` instead of `lastFetched`?

```javascript
if (docsRuleCache.has(rule)) {
  const cachedRule = docsRuleCache.get(rule);
  if (cachedRule.expires >= new Date().getTime()) {
    // If documentation is stale, clear cache
    docsRuleCache.delete(rule);
  } else {
    return cachedRule.markdown;
  }
}
...
docsRuleCache.set(rule, {
  markdown: markdownOutput,
  expires: new Date().getTime() + DOCUMENTATION_LIFETIME,
});
```

Reply:
Yep, absolutely! That's what I meant by [this comment](https://github.com/AtomLinter/linter-reek/pull/120#discussion_r255318543).

Reply:
Ready! :rocket: